pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -91,6 +91,12 @@ func (a *v1beta1PolicyApplier) setAuthnFilterForPeerAuthn(proxyType model.NodeTy
 		return config
 	}
 
+	if proxyType == model.SidecarProxy && a.consolidatedPeerPolicy == nil {
```

Comment:
No policy means permissive mode, in which case I think we do still need the filter. We only don't need it if the policy is plaintext?

Reply:
Hm, all the tests pass. Either my understanding is wrong or the tests have a gap. wdyt @yangminzhu ?

Reply:
Actually, I found no clear definition of the behavior when no applicable beta policy exists (particularly when v1alpha1 authn policy is deprecated).

The only reference that can be obtained from the [concept](https://istio.io/latest/docs/concepts/security/#permissive-mode) is - `When the mode is unset, the mode of the parent scope is inherited. Mesh-wide peer authentication policies with an unset mode use the PERMISSIVE mode by default.`. But this is not exactly equal to `no policy means permissive mode`.

Reply:
IIRC, `consolidatedPeerPolicy` is set to permissive if nothing exists at this line during constructor.

https://github.com/istio/istio/blob/be8d7d33ae674da390f4b850ba2dd0b948cbf7af/pilot%2Fpkg%2Fsecurity%2Fauthn%2Fv1beta1%2Fpolicy_applier.go#L496

So I suppose you only hit this line condition when there's explicit DISABLE to be in plain text. But just to double check, you can run an build and default installation(no beta policy exists) to double check. And confirm the config dump still sets two filter chains plus the authn filter.

Reply:
@incfly Thanks for pointing this out!

However - shouldn't this fall into the return of `nil` condition firstly, before defaulting to permissive if nothing exists?
https://github.com/istio/istio/blob/be8d7d33ae674da390f4b850ba2dd0b948cbf7af/pilot%2Fpkg%2Fsecurity%2Fauthn%2Fv1beta1%2Fpolicy_applier.go#L489-L493

I've tested with the default installation, the authn filter will be set as below.
`
                                {
                                    "name": "istio_authn",
                                    "typedConfig": {
                                        "@type": "type.googleapis.com/istio.envoy.config.filter.http.authn.v2alpha1.FilterConfig",
                                        "policy": {
                                            "peers": [
                                                {
                                                    "mtls": {
                                                        "mode": "PERMISSIVE"
                                                    }
                                                }
                                            ]
                                        },
                                        "skipValidateTrustDomain": true
                                    }
                                },
`
But I don't think that's made by the constructor. It should be produced by `setAuthnFilterForPeerAuthn()` instead.

I can certainly fix the original issue by following that we only don't need the authn filter if the policy is (explicitly) plaintext. Anyway, from the concept level (rather than the implementation level), should we clarify/confirm the behavior when no applicable beta policy exists? Does it make any sense/usage if being defaulted to permissive (and tests should cover that case too)?

Reply:
`getMutualTLSModeForPort()` will return permissive if `consolidatedPeerPolicy == nil` so I think you could just return nil if `effectiveMTLSMode == MTLSDisable`.


> I've tested with the default installation, the authn filter will be set as below.
{ "name": "istio_authn", "typedConfig": { "@type": "type.googleapis.com/istio.envoy.config.filter.http.authn.v2alpha1.FilterConfig", "policy": { "peers": [ { "mtls": { "mode": "PERMISSIVE" } } ] }, "skipValidateTrustDomain": true } },
But I don't think that's made by the constructor. It should be produced by setAuthnFilterForPeerAuthn() instead.

Hmm, did you double check that you have deleted all PeerAuthN and you are also testing the istiod built from this PR? 

We should add an e2e test in reachability_test.go that would fail with the original change in this PR, it tests the default permissive mode when there are no PeerAuthN applied.

Reply:
Thanks for the comments!

> `getMutualTLSModeForPort()` will return permissive if `consolidatedPeerPolicy == nil` so I think you could just return nil if `effectiveMTLSMode == MTLSDisable`.
> 
Agree, this should be the right place where the permissive authn filter is generated when nothing exists.

> > I've tested with the default installation, the authn filter will be set as below.
> > { "name": "istio_authn", "typedConfig": { "@type": "type.googleapis.com/istio.envoy.config.filter.http.authn.v2alpha1.FilterConfig", "policy": { "peers": [ { "mtls": { "mode": "PERMISSIVE" } } ] }, "skipValidateTrustDomain": true } },
> > But I don't think that's made by the constructor. It should be produced by setAuthnFilterForPeerAuthn() instead.
> 
> Hmm, did you double check that you have deleted all PeerAuthN and you are also testing the istiod built from this PR?
> 
Well... I was talking about the current behavior with default installation (i.e. *NOT* the istiod built from this PR)... Sorry for the misleading...

> We should add an e2e test in reachability_test.go that would fail with the original change in this PR, it tests the default permissive mode when there are no PeerAuthN applied.

Thanks for the clarification. This makes sense to me.